### ISSUE TITLE
Reproducing paper results

### ISSUE THREAD

**Issue by cattaneod** [PRBonn/OverlapNet#9]
Dead authors,

Thank you for your work!
I'm trying to reproduce the results reported in the paper using the provided pre-trained model.
I generated the preprocessed data and the ground truths using the scripts demo1 and demo 4, and I'm testing the network with the testing.py file.

Since the pre-trained model use only depths and normals, I expected to obtain a mean rotation error of ~2.97° (as reported in Table V).
However, I'm getting a mean error of 8.90°.

Generated histograms:
![image](https://user-images.githubusercontent.com/11426917/94904827-337a6280-049c-11eb-8afc-fa1445677c0d.png)
![image](https://user-images.githubusercontent.com/11426917/94904837-37a68000-049c-11eb-9732-81495fa0bb8f.png)

PS: To generate the ground truths I'm using the poses provided with the [semantic KITTI dataset](http://semantic-kitti.org/)

**Comment by Chen-Xieyuanli** [comment]
Hey @cattaneod,

Thank you for following our work!

The results of Table IV and Table V are generated following the metric introduced in OEROS paper where the error of rotation is not the average over all possible pairs but with the following setup:

"In case of the KITTI dataset collection, only Sequence 00 revisits the same places again, and can thus be used for proper localization evaluation. Sequences 03-08 are used for training, while Sequence 02 is used for validation. For the evaluation, point clouds from the first 170s of Sequence 00 are used to generate the map, i.e., to populate the KD-tree. The remaining point clouds are used for localization queries. This split of Sequence 00 prevents any self-localization, as the vehicle starts to revisit previously traversed areas after 170s. Analogous to the NCLT datasets, the query point clouds are sampled to be at least 3m apart."

For more information about the metric, I would like to refer you to the original paper: https://arxiv.org/pdf/1903.07918.pdf.

I hope this helps!

**Comment by Chen-Xieyuanli** [comment]
Hey @cattaneod,

I will appreciate it a lot if you could also post your result plots as images inside this GitHub issue.

I think that would be more convenient for other authors and users to following up on this issue.

Thanks!

**Comment by cattaneod** [comment]
Thank you for your quick reply.

I updated my original comment to include the images directly in the comment.

**Comment by cattaneod** [comment]
From what i understand, the steps should be:
1 - select a subset of the KITTI 00 sequence, so that each point cloud is at least 3m apart from the previous one.
2 - use the frames of the first 170s of the sequence to build a KD-Tree
3 - for every remaining frame, compute the yaw between the selected frame and the closest frame in the KD-tree.

Is that correct? I'm not sure about the step 3, it is not completely clear from the OREOS paper.

EDIT: Is it possible for you to share the code for this evaluation? that would greatly help me.

**Comment by Chen-Xieyuanli** [comment]
> From what i understand, the steps should be:
> 1 - select a subset of the KITTI 00 sequence, so that each point cloud is at least 3m apart from the previous one.
> 2 - use the frames of the first 170s of the sequence to build a KD-Tree
> 3 - for every remaining frame, compute the yaw between the selected frame and the closest frame in the KD-tree.
> 
> Is that correct? I'm not sure about the step 3, it is not completely clear from the OREOS paper.
> 
> EDIT: Is it possible for you to share the code for this evaluation? that would greatly help me.

Sorry for the late reply.

Yes, that's the metric from the OREOS paper. I don't think we have the copyright to release the code of this metric. You could write an email to the original author, Lucas (lukas.schaupp@mavt.ethz.ch), asking for the code.

**Comment by Chen-Xieyuanli** [comment]
Hey @cattaneod,

Is there any update on this issue?

**Comment by Chen-Xieyuanli** [comment]
Since there is no update for this issue, I'm going to close it.

If there is any problem, please feel free to ask me to reopen it.
